app/controllers/autocomplete_controller.rb

```diff
@@ -88,18 +88,36 @@ def associated_tags
 
   ## NONCANONICAL TAGS
   def noncanonical_tag
-    search_param = params[:term]
+    search_param = Query.new.escape_reserved_characters(params[:term])
     raise "Redshirt: Attempted to constantize invalid class initialize noncanonical_tag #{params[:type].classify}" unless Tag::TYPES.include?(params[:type].classify)
+
     tag_class = params[:type].classify.constantize
-    render_output(tag_class.by_popularity
-                      .where(["canonical = 0 AND name LIKE ?",
-                              '%' + search_param + '%']).limit(10).map(&:name))
+    one_tag = tag_class.find_by(canonical: false, name: search_param)
+    # Is there a tag which is just right ( this is really for testing )
```

Comment:
```suggestion
    # Is there a tag which is just right (this is really for testing)
```

(The spaces inside are right for code! Just not so much comments 😆)